src/mbgl/map/map.cpp

```diff
@@ -997,6 +1004,22 @@ void Map::Impl::onUpdate(Update flags) {
     
 void Map::Impl::onStyleLoaded() {
     view.notifyMapChange(MapChangeDidFinishLoadingStyle);
+    
+    if (initialCameraOptions == nullptr) {
+        // user manipulated camera before style was loaded
+        return;
+    }
+    
+    // move camera to default camera of style
+    initialCameraOptions->center = style->getDefaultLatLng();
+    initialCameraOptions->zoom = style->getDefaultZoom();
+    initialCameraOptions->angle = style->getDefaultBearing();
```

Comment:
`getDefaultBearing()` should probably stay in degrees clockwise from north. Conventionally, public-facing APIs like `mbgl::Map` use degrees, while internal classes like `TransformState` and `Transform` use radians. However, `CameraOptions` is passed around up and down the stack, and it’s in radians. It would be a very large project to make all of mbgl use radians, so for this PR, I think you should convert to radians right here and perhaps document `getDefaultBearing()`’s current behavior.